src/igraph/drawing/__init__.py

```diff
@@ -482,7 +490,7 @@ def plot(obj, target=None, bbox=(0, 0, 600, 600), *args, **kwds):
     # so just show or save the result
     if target is None:
         result.show()
-    elif isinstance(target, basestring):
```

Comment:
This breaks compatibility with Python 2.7 where filenames could be instances of `bytes` or `str`, but I think we will drop support for 2.7 anyway soon so I don't mind. The same applies to occurrences of `xrange`.